File: backend/app/schemas/professional.py
import uuid
from typing import Optional
from pydantic import BaseModel


class ProfessionalResponse(BaseModel):
    id: uuid.UUID
    professional_type: str
    business_name: Optional[str] = None
    bio: Optional[str] = None
    is_verified: bool
    average_rating: float
    review_count: int
    accepts_new_orders: bool

    model_config = {"from_attributes": True}

File: frontend/src/types/professional.ts
export type ProfessionalType = "tailor" | "designer";

export interface Professional {
  id: string;
  user_id: string;
  professional_type: ProfessionalType;
  business_name?: string | null;
  bio?: string | null;
  years_experience?: number | null;
  shop_photo_url?: string | null;
  is_verified: boolean;
  average_rating: number;
  review_count: number;
  accepts_new_orders: boolean;
}
